### ISSUE TITLE
Missing flatMap (and some other new array methods)

### ISSUE THREAD

**Issue by Qix-** [adamhaile/S-array#13]
Hello again :) In newer environments there are methods such as `.flatMap` and whatnot - could these possibly be supported?

**Comment by adamhaile** [comment]
Yeah, I don't see why not.  If you want to try a PR go for it, if not, I'll try and hit it on the next version.
